Updated README (#4)
thought-machine/phorge: README.md

> # Phorge
> *ThoughtMachine.* Most of the changes we've
> made for ourselves over the years
> except for a few we no longer need
> or just don't fit in style wise, but
> including some that are hard to find
> because either people forgot about
> them or simply because they haven't
> been released yet, a few we really love,
> one we think is just ok, some we did
> for free, some we did for money, some
> for ourselves without permission and
> some for friends as swaps but never on
> time and always at our office in ~~Old Street~~ Herbrand Street.
> 

Reviewer: ```suggestion
```

Reviewer 2: Awww, but that bit is a fun callback to [the original fork](https://github.com/thought-machine/phabricator) :(

# About this repository

Thought Machine's fork of [Phorge](https://github.com/phorgeit/phorge). Phorge itself is a community-maintained

> fork of [Phabricator](http://phabricator.org), developed and maintained by [The Phorge Team](https://phorge.it).

Reviewer: ```suggestion
fork of [Phabricator](http://phabricator.org), developed and maintained by [the Phorge team](https://phorge.it).
```

Our changes are maintained on the `tm-master` branch.

We probably won't accept any pull requests for new features as this repository
and the associated changes are purely quality-of-life changes for our own
installation. Changes that improve the style or robustness will probably be

> accepted though!

Reviewer: I don't think we should encourage this unless the style or robustness improvements are for our own code - we want to keep this as close to upstream as possible, so those changes should go upstream.

We likewise provide no guarantee you'll find the changes useful for your own
Phorge/Phabricator install.

However, we have documented the changes here so you can make up your own mind.

# Overview of Changes

> An list of changes are below; changes originally created in thought-machine/phabricator are marked with an asterix:

Reviewer: ```suggestion
Changes originally created in [thought-machine/phabricator](https://github.com/thought-machine/phabricator) are marked with *:
```

   * Enable bots to add Jira tickets to diffs.
 * Diffusion
   * Added Git upload archive workflow. (*)
   * Added prometheus metrics to track cluster sync successes and failures. (*)
   * Added herald rules and harbourmaster builds for refs. (*)
 * Drydock
   * Modified the working copy blueprint to rebase before trying to merge. (*)
 * Prometheus
   * Added prometheus metric infrastructure and application. (*)
   * Added phabricator up metric to indicate whether phabricator is up or not. (*)
 * People
   * *Change* to UserQuery conduit method
     * We always return the user's email as it is not private information in
       our organisation. (*)
 * Project
   * *Change* to ProjectBoardView controller
     * We allow this page to be `frameable` so we can embed it in other dashboards. (*)
   * *Change* to ProjectBoardTaskCard
     * We display the current status on the task.
   * Add project logical viewer for project or user function so we can do viewerprojects() in queries. (*)
 * Repository
   * Pointing clone URIs to the Phabricator URI even if a repo is hosted elsewhere. (*)
   * added buildable interface for push log for harbormaster builds for refs. (*)

> *Note*: We previously maintained our changes in a separate repository, [thought-machine/phabricator](https://github.com/thought-machine/phabricator); all changes were copied over [upon creation](https://github.com/thought-machine/phorge/commit/2409b3c90d4b896c83dd09eb7d570caba7126cd6) of this repository. To see the changes in chronological order, please see the [commit history](https://github.com/thought-machine/phabricator/commits/master/) of thought-machine/phabricator. Changes are marked in the source files with "TM CHANGES".

Reviewer: Please add line breaks to this to make it readable.

# Installation Instructions

To install this repository, use:

```
git clone --single-branch --branch tm-master https://github.com/thought-machine/phorge tmphorge
```

# Dependencies

> We use [composer](https://getcomposer.org/) to manage external dependencies. [This](https://getcomposer.org/doc/01-basic-usage.md) page gives more information on how to use it.

Reviewer: Line breaks here too, please.
